tests/test_backups.py

```diff
@@ -0,0 +1,171 @@
+# SPDX-FileCopyrightText: Copyright DB Netz AG and the capella-collab-manager contributors
+# SPDX-License-Identifier: Apache-2.0
+
+import logging
+import os
+import pathlib
+import subprocess
+
+import conftest
+import pytest
+from docker.models import containers
+
+log = logging.getLogger(__file__)
+log.setLevel("DEBUG")
+
+CAPELLA_VERSION = os.getenv("CAPELLA_VERSION", "")
+
+
+def is_capella_5_x_x() -> bool:
+    return CAPELLA_VERSION in ("5.0.0", "5.2.0")
+
+
+def is_capella_6_x_x() -> bool:
+    return CAPELLA_VERSION in ("6.0.0")
+
+
+@pytest.fixture(name="local_git_container")
+def fixture_local_git_container() -> containers.Container:
+    ports: dict[str, int] = {"80/tcp": 9010}
+
+    with conftest.get_container(
+        image="tests/local-git-server", ports=ports
+    ) as container:
+        conftest.wait_for_container(container, "server started")
+        yield container
+
+
+@pytest.fixture(name="t4c_backup_container")
+def fixture_t4c_backup_container(request) -> containers.Container:
+    env: dict[str, str] = {
+        "GIT_REPO_URL": "http://localhost:9010/git/git-test-repo.git",
+        "GIT_REPO_BRANCH": "backup-test",
+        "GIT_USERNAME": "any",
+        "GIT_PASSWORD": "any",
+        "T4C_REPO_HOST": "127.0.0.1",
+        "T4C_REPO_PORT": "2036",
+        "T4C_CDO_PORT": "12036",
+        "T4C_REPO_NAME": "test-repo",
+        "T4C_PROJECT_NAME": "test-project",
+        "T4C_USERNAME": "admin",
+        "T4C_PASSWORD": "password",
+        "LOG_LEVEL": "INFO",
+        "INCLUDE_COMMIT_HISTORY": "false",
+    } | request.param
+
+    with conftest.get_container(
+        image="t4c/client/backup", environment=env, network="host"
+    ) as container:
+        yield container
+
+
+@pytest.fixture(name="t4c_server_container")
+def fixture_t4c_server_container(request) -> containers.Container:
+    env: dict[str, str] = {"REST_API_PASSWORD": "password"} | request.param
+
+    ports: dict[str, int] = {
+        "2036/tcp": 2036,
+        "8080/tcp": 8081,
+        "12036/tcp": 12036,
+    }
```

Comment:
I haven't applied it yet because I thought it would conflict with my current test strategy (same problem as the old fixture placement), but since the problem is resolved, I can now use the random port assignment. So I applied this.